lua/mLRS.lua

```diff
@@ -1183,44 +1215,86 @@ local function drawPageMain()
+            lcd.drawText(245, y, tostring(DEVICE_INFO.rx_power_dbm).." dBm", rx_attr)
+            if DEVICE_INFO.rx_diversity <= #diversity_list then
+                lcd.drawText(245, y+15, diversity_list[DEVICE_INFO.rx_diversity], rx_attr)
+            else
+                lcd.drawText(245, y+15, "?", rx_attr)
+            end
         else
-            lcd.drawText(140+240, y+20, "?", rx_attr)
+            lcd.drawText(245, y, "---", rx_attr)
+            lcd.drawText(245, y+15, "---", rx_attr)
+        end
+        
+        y = y + 30
+        lcd.drawText(10, y, "Sensitivity", g_textColor)
+        if DEVICE_INFO ~= nil then
+            lcd.drawText(90, y, tostring(DEVICE_INFO.receiver_sensitivity).." dBm", g_textColor)
+        else
+            lcd.drawText(90, y, "---", g_textColor)
         end
     else
-        lcd.drawText(140+240, y, "---", rx_attr)
-        lcd.drawText(140+240, y+20, "---", rx_attr)
-    end
+        lcd.drawText(10, y, "Tx Power", g_textColor)
+        lcd.drawText(10, y+20, "Tx Diversity", g_textColor)
+        if DEVICE_INFO ~= nil then
+            lcd.drawText(140, y, tostring(DEVICE_INFO.tx_power_dbm).." dBm", g_textColor)
+            if DEVICE_INFO.tx_diversity <= #diversity_list then
+                lcd.drawText(140, y+20, diversity_list[DEVICE_INFO.tx_diversity], g_textColor)
+            else
+                lcd.drawText(140, y+20, "?", g_textColor)
+            end
+        else
+            lcd.drawText(140, y, "---", g_textColor)
+            lcd.drawText(140, y+20, "---", g_textColor)
+        end
 
-    y = y + 2*20
-    lcd.drawText(10, y, "Sensitivity", g_textColor)
-    if DEVICE_INFO ~= nil then
-        lcd.drawText(140, y, tostring(DEVICE_INFO.receiver_sensitivity).." dBm", g_textColor)
-    else
-        lcd.drawText(140, y, "---", g_textColor)
+        local rx_attr = g_textColor
+        if not connected then
+            rx_attr = g_textDisableColor
+        end
+        lcd.drawText(10+LCD_W_HALF, y, "Rx Power", rx_attr)
+        lcd.drawText(10+LCD_W_HALF, y+20, "Rx Diversity", rx_attr)
+        if DEVICE_INFO ~= nil and connected then
+            lcd.drawText(140+LCD_W_HALF, y, tostring(DEVICE_INFO.rx_power_dbm).." dBm", rx_attr)
+            if DEVICE_INFO.rx_diversity <= #diversity_list then
+                lcd.drawText(140+LCD_W_HALF, y+20, diversity_list[DEVICE_INFO.rx_diversity], rx_attr)
+            else
+                lcd.drawText(140+LCD_W_HALF, y+20, "?", rx_attr)
+            end
+        else
+            lcd.drawText(140+LCD_W_HALF, y, "---", rx_attr)
+            lcd.drawText(140+LCD_W_HALF, y+20, "---", rx_attr)
+        end
+
+        y = y + 2*20
+        lcd.drawText(10, y, "Sensitivity", g_textColor)
+        if DEVICE_INFO ~= nil then
+            lcd.drawText(140, y, tostring(DEVICE_INFO.receiver_sensitivity).." dBm", g_textColor)
+        else
+            lcd.drawText(140, y, "---", g_textColor)
+        end
```

Comment:
hm, really lots of code duplication
can we do that generically by a set of values for positions?